R/testdata.R

```diff
@@ -253,19 +253,36 @@ dataset_test_worker <-
       expect(
         is.null(error),
         sprintf("%s threw an error:\n\n" %+% red("%s"), info, paste(error, collapse = ",")))
-      invisible(NULL)
+      invisible(object)
+    }
+
+    expect_no_warning <- function(object, ..., info) {
+      warning <- tryCatch({
+        object
+        NULL
+      }, warning = function(w) {
+        w
+      })
+      expect(is.null(warning), info)
     }
 
     expect_list_elements_contains_names <- function(object, expected, info) {
       for (i in seq_along(object))
         expect_contains(names(object[[i]]), expected, info = paste(info, i))
-      invisible(NULL)
+      invisible(object)
     }
 
     expect_list_elements_allowed_names <- function(object, allowed, info, label) {
       for (i in seq_along(object))
         expect_allowed(names(object[[i]]), allowed, info = paste(info, i), label = "field names")
-      invisible(NULL)
+      invisible(object)
+    }
+
+    expect_list_elements_exact_names <- function(object, expected, info) {
+      for (i in seq_along(object))
+        expect_contains(names(object[[i]]), expected, info = paste(info, i))
+        expect_allowed(names(object[[i]]), expected, info = paste(info, i), label = "field names")
+      invisible(object)
```

Comment:
I don't know why these changes are necessary - can you explain?

Reply:
It's not super necessary but it was to make the code more compact. Instead of doing both `expect_list_elements_contains_names` and `expect_list_elements_allowed_names` on the same object, now you just run one function (`expect_list_elements_exact_names`)